shpc/main/modules/base.py

```diff
@@ -269,18 +264,17 @@ def _cleanup_symlink(self, module_dir):
         symlinked_module = self.get_symlink_path(module_dir)
         if not symlinked_module:
             return
-        if os.path.exists(symlinked_module) and os.path.islink(symlinked_module):
+        if os.path.islink(symlinked_module):
             os.unlink(symlinked_module)
-
-        # Clean up directories that become empty
-        parent_dir = os.path.dirname(symlinked_module)
-        if not os.path.exists(parent_dir):
-            return
-
-        # If the parent of the symlink only has zero files OR one file .version, cleanup
-        files = os.listdir(parent_dir)
-        if len(files) == 0 or (len(files) == 1 and files[0] == ".version"):
-            shutil.rmtree(parent_dir)
+            # Clean up directories that become empty
+            utils.rmdir_to_base(symlinked_module, self.settings.symlink_base)
+            logger.info("%s has been removed." % symlinked_module)
+            # Update .version
+            self.write_version_file(os.path.dirname(symlinked_module))
+        elif os.path.exists(symlinked_module):
+            logger.error("%s exists and is not a symlink!" % symlinked_module)
+        elif self.settings.symlink_tree:
```

Comment:
For this last case - the implication is "symlink_module doesn't exist but we have a tree so there must be an issue?" Could we add a comment here so future me can remember :)